src/main/java/io/kestra/plugin/docker/Build.java

```diff
@@ -0,0 +1,218 @@
+
+    @Schema(
+        title = "The target platform for the image e.g. linux/amd64."
+    )
+    @PluginProperty(dynamic = true)
+    private List<String> platforms;
+
+    @Schema(
+        title = "Whether to push the image to a remote container registry."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean push = false;
+
+    @Schema(
+        title = "Always attempt to pull a newer version of the base image."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean pull = true;
+
+    @Schema(
+        title = "The tag of this image."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String tag;
+
+    @Schema(
+        title = "Optional build arguments in a `key: value` format."
+    )
+    @PluginProperty(
+        additionalProperties = String.class,
+        dynamic = true
+    )
+    protected Map<String, String> buildArgs;
+
+    @Schema(
+        title = "Additional metadata for the image in a `key: value` format."
+    )
+    @PluginProperty(
+        additionalProperties = String.class,
+        dynamic = true
+    )
+    protected Map<String, String> labels;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        DefaultDockerClientConfig.Builder builder = DefaultDockerClientConfig.createDefaultConfigBuilder()
+            .withDockerHost(DockerService.findHost(runContext, this.host));
+
+        try (DockerClient dockerClient = DockerService.client(builder.build())) {
+            BuildImageCmd buildImageCmd = dockerClient.buildImageCmd()
+                .withPull(this.pull);
+
+            Path path = runContext.tempDir();
+            String dockerfile = runContext.render(this.dockerfile);
+            Path dockerFile;
+
+            if (path.resolve(dockerfile).toFile().exists()) {
+                dockerFile = path.resolve(dockerfile);
+            } else {
+                dockerFile = runContext.tempFile(dockerfile.getBytes(StandardCharsets.UTF_8), ".dockerfile");
+            }
+
+            buildImageCmd.withDockerfile(dockerFile.toFile());
+
+            if (this.platforms != null) {
+                runContext.render(this.platforms).forEach(buildImageCmd::withPlatform);
+            }
+
+            if (this.tag != null) {
+                buildImageCmd.withTags(Set.of(runContext.render(this.tag)));
+            }
+
+            if (this.buildArgs != null) {
+                runContext.renderMap(this.buildArgs).forEach(buildImageCmd::withBuildArg);
+            }
+
+            if (this.labels != null) {
+                buildImageCmd.withLabels(runContext.renderMap(this.labels));
+            }
+
+            String imageId = buildImageCmd
+                .exec(new BuildImageResultCallback())
+                .awaitImageId();
+
+            if (this.push) {
+                AuthConfig authConfig = new AuthConfig()
+                    .withRegistryAddress(DockerService.registryUrlFromImage(tag));
```

Comment:
Maybe it would be better to ask for the registry URL as a property instead of trying to extract it from the tag.

Reply:
done here if it's defined: https://github.com/kestra-io/plugin-docker/blob/a3f401c462cbb00e424b939b0cd9183cd6784878/src/main/java/io/kestra/plugin/docker/Build.java#L172